lib/utils/visUtil.cpp

```diff
@@ -341,19 +347,52 @@ double StatTracker::get_min() {
 }
 
 double StatTracker::get_avg() {
+    std::lock_guard<std::mutex> lock(tracker_lock);
+
     if (count == 0) {
         return NAN;
     }
     return avg;
 }
 
 double StatTracker::get_std_dev() {
+    std::lock_guard<std::mutex> lock(tracker_lock);
+
     if (count <= 1) {
         return NAN;
     }
     return std_dev;
 }
 
+nlohmann::json StatTracker::get_json() {
+    std::lock_guard<std::mutex> lock(tracker_lock);
+
+    nlohmann::json tracker_json = {};
+    tracker_json["unit"] = unit;
+    for (size_t i = 0; i < count; i++) {
+        nlohmann::json sample_json = {};
+        sample_json["timestamp"] = std::chrono::duration_cast<std::chrono::milliseconds>(
+                                       rbuf[i].timestamp.time_since_epoch())
+                                       .count();
+        sample_json["value"] = rbuf[i].value;
+        tracker_json["samples"].push_back(sample_json);
+    }
+
+    return tracker_json;
+}
+
+nlohmann::json StatTracker::get_current_json() {
+    nlohmann::json tracker_json = {};
+
+    tracker_json["unit"] = unit;
+    tracker_json["min"] = get_min();
```

Comment:
We should add the current value, and timestamp of the current value to this as well.